### PR TITLE
Chore: Add new CloudWatch ui components

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:
Adds new UI components that will be used by the CloudWatch data source plugin. 

**Special notes for your reviewer**:
These components will be made available (in some form) to other plugins developers eventually. When doing so, storybook stories should be created for these components.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/ui/EditorList.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const onDeleteItem = (itemIndex: number) => {
+    const newItems = [...items];
+    newItems.splice(itemIndex, 1);
+    onChange(newItems);
+  };
+
+  return (
+    <Stack>
+      {items.map((item, index) => (
+        <div key={index}>
```
Comment (joshhunt) [inline public/app/plugins/datasource/cloudwatch/components/ui/EditorList.tsx:33]: I know I made this 😂  - but looking at it again again if possible, i think `<Fragment key={index}`/>` would be cleaner, if possible.

There's the possibility this causes regressions in the few places its used, so I wouldn't stress too much about it.

**Review by joshhunt — APPROVED** [review]

**Comment by dprokop** [comment]
@sunker are those the components used by the new visual query builder? We would like to use some of those in BQ visual query builder and were thinking about creating a separate package (@grafana/query-builder?). cc @zoltanbedi

**Comment by sunker** [comment]
> @sunker are those the components used by the new visual query builder? We would like to use some of those in BQ visual query builder and were thinking about creating a separate package (@grafana/query-builder?). cc @zoltanbedi

Correct! Yes, next step is to make these components available to other plugin devs. I also think an npm package is, at least initially, the right path forward, but let's discuss this with the @grafana/plugins-platform. I'll send an invite for next week.

File: public/app/plugins/datasource/cloudwatch/components/ui/InlineSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    ValueContainer,
+    SingleValue: ValueContainer,
+  };
+
+  return (
+    <div className={styles.root}>
+      {labelProp && (
+        <label className={styles.label} htmlFor={id}>
+          {labelProp}
+          {':'}&nbsp;
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloudwatch/components/ui/InlineSelect.tsx:31]: why not using a margin-right for the label?

File: public/app/plugins/datasource/cloudwatch/components/ui/InputGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      '&:last-child': {
+        borderTopLeftRadius: 0,
+        borderBottomLeftRadius: 0,
+      },
+
+      '&:not(:first-child):not(:last-child)': {
+        borderRadius: 0,
+      },
+
+      //
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloudwatch/components/ui/InputGroup.tsx:40]: leftover comment

**Review by andresmgot — APPROVED** [review]
LGTM

**Pull request merged**